pkg/provider/identity.go

```diff
@@ -35,26 +54,235 @@ type PeerIdentity struct {
 type FullIdentity struct {
 	PeerIdentity
 	PrivateKey crypto.PrivateKey
+}
 
-	todoCert *tls.Certificate // TODO(jt): get rid of this and only use the above
+type CAConfig struct {
+	CertPath   string `help:"path to the certificate chain for this identity" default:"$CONFDIR/identity.leaf.cert"`
+	KeyPath    string `help:"path to the private key for this identity" default:"$CONFDIR/identity.leaf.key"`
+	Difficulty uint16 `default:"24" help:"minimum difficulty for identity generation"`
```

Comment:
once a CA is created, you don't need the difficulty anymore? difficulty seems like a creation thing, not a property of an existing CA